zplug: update the output path

15cm opened the pull request:
This is a breaking change because the old behavior pollutes the nix profile root dir with all files in https://github.com/zplug/zplug and needs to be fixed.

I created a corresponding PR in the home manager repo https://github.com/nix-community/home-manager/pull/3922. For non HM users, they will need to update their dependency on `${pkgs.zplug}/init.zsh` to `${pkgs.zplug}/share/zplug/init.zsh`.

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

15cm committed:
zplug: update the output path

This is a breaking change because the old behavior pollutes the nix profile root
dir with all files in https://github.com/zplug/zplug and needs to be fixed.

I created a corresponding PR in the home manager repo
https://github.com/nix-community/home-manager/pull/3922. For non HM users, they
will need to update their dependency on `${pkgs.zplug}/init.zsh` to `${pkgs.zplug}/share/zplug/init.zsh`.

ncfavier commented on pkgs/shells/zsh/zplug/default.nix:
```diff
@@ -17,8 +17,8 @@ stdenv.mkDerivation rec {
   installPhase = ''
-    mkdir -p $out
-    cp -r $src/* $out/
+    mkdir -p $out/share/zplug
+    cp -r $src/* $out/share/zplug/
```
Let's at least rescue `bin` to `/bin` and `doc/man` to `/share/man`...

15cm replied:
Good point. Done. `bin` is added to `/share/zplug/bin` instead of `/bin` because it will be added to `$PATH` via `$ZPLUG_ROOT/bin` in [zplug//base/core/core.zsh](https://github.com/zplug/zplug/blob/ac6c2a3e9eea6a488d96d98c752ef887e7d5aae3/base/core/core.zsh#L135).

I refer to the [AUR zplug package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zplug) for file list to copy.

15cm committed:
Only add necessary files to $out/share/zplug. Also add the zplug man pages

The LICENSE file is not in the 2.4.2 tag and there's not a release after that. I
would skip adding the license $out/licenses/zplug in this commit.

Reference: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zplug
